### PR TITLE
Add getMinimal method

### PR DESCRIPTION
**Pull request by drklee3**
Adds a faster getMinimal method to fetch only the ID, displayName, phoneNumbers, and thumbnailPath of contacts via `ContactsContract.CommonDataKinds.Phone` instead of `ContactsContract.Profile`.

### CODE REVIEW

File: android/src/main/java/com/rt2zz/reactnativecontacts/ContactsProvider.java [hunk 1]
```diff
@@ -170,6 +179,78 @@ public WritableMap getContactById(String contactId) {
        return null;
     }
 
+    public WritableArray getMinimalContacts() {
```
Comment (jmrossy) [inline android/src/main/java/com/rt2zz/reactnativecontacts/ContactsProvider.java:227]: Let's break up this function a bit

File: example/android/app/src/main/java/com/example/MainApplication.java [hunk 2]
```diff
@@ -3,6 +3,9 @@
 import android.app.Application;
 
 import com.facebook.react.ReactApplication;
+import com.segment.analytics.reactnative.core.RNAnalyticsPackage;
```
Comment (jmrossy) [inline example/android/app/src/main/java/com/example/MainApplication.java]: Consider removing this analytics package

File: package.json [hunk 3]
```diff
@@ -31,11 +31,16 @@
...
+    {
+      "name": "Nick Pomfret"
+    },
+    {
+      "name": "rt2zz <zack@root-two.com>"
+    }
   ],
   "license": "MIT",
   "dependencies": {
+    "react-native": "^0.59.5",
```
Comment (jmrossy) [inline package.json]: consider moving this to peerDependencies. Unless needed to make example app work, in which case it can go in devDependencies

**Comment by aaronmgdr** [comment]
abandoned?

**Comment by drklee3** [comment]
Currently the branch with `getMinimal()` isn't compatible with master (since master migrated to AndroidX), so the Celo mobile wallet just uses the latest commit on the branch directly (https://github.com/celo-org/react-native-contacts/pull/1/commits/4989b0b4df8bf84acf1979643af243b51453a504).

I guess this PR can be closed for now as migrating to AndroidX will require a few additional changes.

**Pull request closed without merging**